examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -0,0 +1,571 @@
+  // Floating Action Button when it is docked.
+  bool notchEnabled = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Bottom App Bar with FAB location'), 
+        // Add 48dp of space onto the bottom of the appbar.
+        // This gives space for the top-start location to attach to without
+        // blocking the 'back' button.
+        bottom: const PreferredSize(
+          preferredSize: const Size.fromHeight(48.0), 
+          child: const SizedBox(),
+        ),
+      ),
+      body: new SingleChildScrollView(
+        padding: const EdgeInsets.all(16.0),
+        child: controls(context),
+      ),
+      bottomNavigationBar: new MyBottomAppBar(_fabLocationConfigurations[fabLocationIndex].babMode, babColor, notchEnabled),
+      floatingActionButton: _fabShapeConfigurations[fabShapeIndex].fab,
+      floatingActionButtonLocation: _fabLocationConfigurations[fabLocationIndex].fabLocation,
+    );
+  }
+
+  Widget controls(BuildContext context) {
+    return new Column(
+      children: <Widget> [
+        new Text(
+          'Floating action button',
+          style: Theme.of(context).textTheme.title,
+        ),
+        new Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            new Container(width: 96.0,
+              child: const Text('Shape: '),
+            ),
+            new Expanded(child: buildFabShapePicker()),
+          ],
+        ),
+        new Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            new Container(width: 96.0,
+              child: const Text('Location: '),
+            ),
+            new Expanded(child: buildFabLocationPicker()),
+          ],
+        ),
+        const Divider(),
+        new Text(
+          'Bottom app bar options',
+          style: Theme.of(context).textTheme.title,
+        ),
+        buildBabColorPicker(),
+        new CheckboxListTile(
+          title: const Text('Enable notch'),
+          value: notchEnabled,
+          onChanged: (bool value) { setState(() { notchEnabled = value; }); },
+          controlAffinity: ListTileControlAffinity.leading,
+        ),
+      ],
+    );
+  }
+
+  Widget buildFabShapePicker() {
+    final List<DropdownMenuItem<int>> options = <DropdownMenuItem<int>>[];
+    for (int i = 0; i < _fabShapeConfigurations.length; i++) {
+      final _FabShapeConfiguration configuration = _fabShapeConfigurations[i];
+      options.add(
+        new DropdownMenuItem<int>(
+          child: new Text(configuration.name),
+          value: i,
+        ),
+      );
+    }
+    return new SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      primary: false,
+      child: new DropdownButton<int>(items: options, onChanged: (int newIdx) { setState(() { fabShapeIndex = newIdx; }); }, value: fabShapeIndex),
```

Comment:
This would be easier to ready like (and we don't use abbreviations like newIdx)
```
child: new DropdownButton<int>(
  items: options,
  onChanged: (int newIndex) {
    setState(() { fabShapeIndex = newIndex; });
  },
  value: fabShapeIndex
),
```

Reply:
Done